backend/file-system/src/main/java/com/htec/filesystem/service/ShelfService.java

```diff
@@ -81,4 +81,16 @@ public void softDeleteShelf(AuthUser user, List<Long> shelfIds) {
         }
         return dtoShelves;
     }
+
+    @Transactional
+    public void hardDeleteShelf(List<Long> shelfIdList, Long userId) {
+
+        List<ShelfEntity> shelfEntities = shelfRepository.findAllByUserIdAndIdIn(userId, shelfIdList);
+
+        if (!shelfEntities.stream().map(ShelfEntity::getId).collect(Collectors.toList()).containsAll(shelfIdList)) {
```

Comment:
try to replace this with if (!shelfEntities.stream().allMatch(e -> shelfIdList.contains(e.getId()))

Reply:
@markobarackovhtec it won't give us expected result cause considering shelfEntities with ids (1,2) and shelfIdList with ids (1,2,3) method wouldn't throw and exception and it should